core/src/test/java/org/testcontainers/junit/CopyFileToContainerTest.java

```diff
@@ -0,0 +1,39 @@
+package org.testcontainers.junit;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.startupcheck.MinimumDurationRunningStartupCheckStrategy;
+import org.testcontainers.utility.MountableFile;
+import java.io.File;
+import java.time.Duration;
+
+public class CopyFileToContainerTest {
+    private static String folderPath = new File("./src/test/resources/mappable-resource/").getAbsolutePath();
+    private static String containerPath = "/opt";
+    private static String fileName = "test-resource.txt";
+    public static GenericContainer container;
+
+    @Before
+    public void before() {
+        container = new GenericContainer("couchbase:latest")
+            .withStartupCheckStrategy(new MinimumDurationRunningStartupCheckStrategy(Duration.ofSeconds(5)))
+            .withCopyFileToContainer(MountableFile.forHostPath(folderPath), containerPath );
+    }
+
+
+    @Test
+    public void checkFileCopied() {
+        container.start();
```

Comment:
Do we need to split the instantiation/start of the container like this, or could the container just be a field with an `@Rule` annotation for this test? I think I'd prefer the latter, as it's the more idiomatic way.

Reply:
actually, didn't we agree to use `try-with-resources` style? Otherwise, if you have 5 rules in your test and run a single test all of them will start - that sucks :(

Reply:
Maybe it depends? This class has only one test method (and would have only one Rule), which is why I suggested this way. I'm _very_ relaxed about it though; `try-with-resources` would also be fine.

Reply:
What is this try-with-resources? could you please a reference to understand this

Reply:
Sure, see here:
https://github.com/testcontainers/testcontainers-java/blob/609f9a10c3c110be405b9cd96fc340c992b79b3b/core/src/test/java/org/testcontainers/junit/GenericContainerRuleTest.java#L346

Reply:
@bsideup Thank you.. that helps